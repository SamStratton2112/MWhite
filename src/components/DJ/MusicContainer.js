import MusicPlayer from './MusicPlayer';
import mike from '../../data';
import { Container, Row, Col } from 'react-bootstrap';

// Displays artists album covers with links to album page on soundcloud. 
const MusicContainer = () =>{
    const allAlbums = [];
    for(let album in mike.dj.links.audio){
        allAlbums.push(<Col>
            <MusicPlayer album={mike.dj.links.audio[album]} />
        </Col>)
    };

    return(
        <Container className="center ">
            <Row  md={12} lg={3} >
                {allAlbums}
            </Row>
        </Container>
    )
};

export default MusicContainer;