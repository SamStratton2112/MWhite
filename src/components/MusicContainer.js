import MusicPlayer from './MusicPlayer';
import mike from '../data';
import { Container, Row, Col } from 'react-bootstrap';

const MusicContainer = () =>{
    return(
        <Container>
            <Row md={12}>
                <Col>
                    <MusicPlayer album={mike.dj.links.audio.unwind} />
                </Col>
                <Col>
                    <MusicPlayer album={mike.dj.links.audio.sexyTime} />
                </Col>
                <Col>
                    <MusicPlayer album={mike.dj.links.audio.inspiredBy} />
                </Col>
            </Row>
        </Container>
    )
};

export default MusicContainer;