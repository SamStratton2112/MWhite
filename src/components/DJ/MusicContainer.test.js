import MusicContainer from './MusicContainer';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

describe(MusicContainer, () => {
    it("renders without crashing", () => {
        render(<MemoryRouter>
            <MusicContainer />
        </MemoryRouter>);
    });
    it("matches snapshot", () => {
        const { asFragment } = render(<MemoryRouter>
            <MusicContainer />
        </MemoryRouter>);
        expect(asFragment()).toMatchSnapshot();
    });
});