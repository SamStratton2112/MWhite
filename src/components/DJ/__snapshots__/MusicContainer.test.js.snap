// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MusicContainer matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="center  container"
  >
    <div
      class="row row-cols-md-12"
    >
      <div
        class="col"
      >
        <div
          class="center card"
          style="color: white; margin: 1em; border: solid; border-color: rgb(184, 17, 17);"
        >
          <img
            alt="Unwind"
            class="card-img-top"
            src="https://i.imgur.com/WAey4uy.jpeg"
            style="width: 15em; height: 16em; padding-top: 1em;"
          />
          <div
            class="card-body"
          >
            <div
              class="card-title h5"
              style="font-size: 1.7em;"
            >
              Unwind
            </div>
            <a
              href="https://soundcloud.com/user-554826787/unwind"
            >
              <button
                class="bttn"
              >
                Listen on Soundcloud
              </button>
            </a>
          </div>
        </div>
      </div>
      <div
        class="col"
      >
        <div
          class="center card"
          style="color: white; margin: 1em; border: solid; border-color: rgb(184, 17, 17);"
        >
          <img
            alt="Sexy Time"
            class="card-img-top"
            src="https://i.imgur.com/VMQqEA7.jpeg"
            style="width: 15em; height: 16em; padding-top: 1em;"
          />
          <div
            class="card-body"
          >
            <div
              class="card-title h5"
              style="font-size: 1.7em;"
            >
              Sexy Time
            </div>
            <a
              href="https://soundcloud.com/user-554826787/sexytime"
            >
              <button
                class="bttn"
              >
                Listen on Soundcloud
              </button>
            </a>
          </div>
        </div>
      </div>
      <div
        class="col"
      >
        <div
          class="center card"
          style="color: white; margin: 1em; border: solid; border-color: rgb(184, 17, 17);"
        >
          <img
            alt="An inspired album..."
            class="card-img-top"
            src="https://i.imgur.com/fHMsQky.jpeg"
            style="width: 15em; height: 16em; padding-top: 1em;"
          />
          <div
            class="card-body"
          >
            <div
              class="card-title h5"
              style="font-size: 1.7em;"
            >
              An inspired album...
            </div>
            <a
              href="https://soundcloud.com/user-554826787/an-album-inspired-by-kruder-dorfmeister-mixed-by-dj-mike-white"
            >
              <button
                class="bttn"
              >
                Listen on Soundcloud
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
